cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -61,6 +85,30 @@ size_t getContingencyMatrixWorkspaceSize(int nSamples,
     nSamples, groundTruth, stream, minLabel, maxLabel);
 }
 
+/**
+ * @brief Calculate workspace size for running contingency matrix calculations
+ * @tparam T label type
+ * @tparam OutT output matrix type
+ * @param handle: the raft handle.
+ * @param groundTruth: device 1-d array for ground truth (num of rows)
+ * @param minLabel: Optional, min value in input array
+ * @param maxLabel: Optional, max value in input array
+ */
+template <typename T,
+          typename OutT = int,
+          typename IdxType,
+          typename LayoutPolicy,
+          typename AccessorPolicy>
+size_t getContingencyMatrixWorkspaceSize(
+  const raft::handle_t& handle,
+  raft::mdspan<T, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> groundTruth,
+  T minLabel = std::numeric_limits<T>::max(),
+  T maxLabel = std::numeric_limits<T>::max())
```

Comment:
This doesn't seem quite right -- should the `maxLabel`'s default value be `numeric_limits<T>::min()`, so that everything else is greater than it?

Reply:
It's used as a special value in the detail function: if one of those label is `numeric_limits<T>::max()`, then both `minLabels` and `maxLabels` are recomputed.
Instead of using this behaviour on the public API I put this parameter as optional now.